cli/src/wallet.rs

```diff
@@ -1,88 +1,108 @@
-        get_password("Stronghold password", false)?
-    } else {
-        get_password("Stronghold password", !snapshot_path.exists())?
-    };
-    let secret_manager = SecretManager::Stronghold(
-        StrongholdSecretManager::builder()
-            .password(&password)
-            .build(snapshot_path)?,
-    );
 
     let (wallet, account) = if let Some(command) = cli.command {
-        if let WalletCommand::Init(init_parameters) = command {
-            (init_command(secret_manager, storage_path, init_parameters).await?, None)
-        } else if let WalletCommand::Restore { backup_path } = command {
-            (
-                restore_command(secret_manager, storage_path, backup_path, password).await?,
-                None,
-            )
-        } else {
-            let wallet = Wallet::builder()
-                .with_secret_manager(secret_manager)
-                .with_storage_path(&storage_path)
-                .finish()
-                .await?;
-            let mut account = None;
-
-            match command {
-                WalletCommand::Backup { backup_path } => {
-                    backup_command(&wallet, backup_path, &password).await?;
-                    return Ok((None, None));
+        match command {
+            WalletCommand::Init(init_parameters) => {
+                let wallet = init_command(storage_path, snapshot_path, init_parameters).await?;
+                (Some(wallet), None)
+            }
+            WalletCommand::Restore { backup_path } => {
+                let wallet = restore_command(storage_path, snapshot_path, std::path::Path::new(&backup_path)).await?;
+                (Some(wallet), None)
+            }
+            WalletCommand::Backup { backup_path } => {
+                backup_command(storage_path, snapshot_path, std::path::Path::new(&backup_path)).await?;
+                return Ok((None, None));
+            }
+            WalletCommand::ChangePassword => {
+                let wallet = change_password_command(storage_path, snapshot_path).await?;
+                (Some(wallet), None)
+            }
+            WalletCommand::New { alias } => {
+                let (wallet, account) = new_command(storage_path, snapshot_path, alias).await?;
+                (Some(wallet), Some(account))
+            }
+            WalletCommand::SetNode { url } => {
+                let wallet = set_node_command(storage_path, snapshot_path, url).await?;
+                (Some(wallet), None)
+            }
+            WalletCommand::Sync => {
+                let wallet = sync_command(storage_path, snapshot_path).await?;
+                (Some(wallet), None)
+            }
+            WalletCommand::Mnemonic => {
+                mnemonic_command().await?;
+                return Ok((None, None));
+            }
+        }
+    } else {
+        // no command provided, i.e. `> ./wallet`
+        match (storage_path.exists(), snapshot_path.exists()) {
```

Comment:
Seems like there is a lot of duplicated code here. Can you not have separate match statements to dedup?

Reply:
hmmm. There's duplicated code indeed:

Reply:
I meant more that there's only one match that handles various "states" when it should be more like a waterfall of states.

Reply:
Could use a loop around the match, for instance.

Reply:
hmmm, thinking about it

Reply:
it's hard for me to say what to make from your proposal ... maybe @thibault-martinez wants to chime in on it. 
The way I see (which may be absolutely wrong), we have 4 possible situations/states we can run into when starting up the app.
1. `.stronghold` file and `wallet-db` exist -> all good -> log in -> send me all your tokens
2. neither of both exit -> all good ... we can fix that -> guide the user through the setup -> enter account -> send me all tokens
3. `wallet-db` exists, but `.stronghold` file does not -> now that is weird (was .stronghold moved?) something fish is going -> abort the app with error message
4. `.stronghold` exists, but `wallet-db` does not -> seems fixable but it's still strange, because on init we usually create both, so something fishy may be going on ... I decided FOR NOW that we should abort as well.

Reply:
And my problem now is that I cannot bring my logic together with your `loop`ing idea :sweat_smile:

Reply:
The loop would just match to check what state we're in, then perform only the thing to go to the next state.

Reply:
Feel free to push your idea to the branch if you want, and we can discuss it.

Reply:
I took a closer look and changed my mind.